com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/jobs/CleanUpJob.java

```diff
@@ -248,18 +249,27 @@ private void removeProcessLogs(List<String> oldFinishedOperationsIds) {
 
     private void removeOldFiles(Map<String, List<String>> spaceToFileIds) {
         try {
-            int removedOldFiles = fileService.deleteAllByFileIds(spaceToFileIds);
+            Map<String, List<String>> spaceToFileChunkIds = splitAllFilesInChunks(spaceToFileIds);
+            int removedOldFiles = fileService.deleteAllByFileIds(spaceToFileChunkIds);
             LOGGER.info("Deleted MTA files: " + removedOldFiles);
         } catch (FileStorageException e) {
             throw new SLException(e);
         }
     }
 
+    public Map<String, List<String>> splitAllFilesInChunks(Map<String, List<String>> spaceToFileIds) {
+        Map<String, List<String>> spaceToFileChunks = new HashMap<String, List<String>>();
+        for (String space : spaceToFileIds.keySet()) {

```

Comment:
Is there a way to achieve this using a Stream::map function? It will be more clear. If not - leave it as it is :)

Reply:
In my opinion it will become very difficult to understand and also difficult to implement.